cmake/kokkos_enable_options.cmake

```diff
@@ -58,6 +58,7 @@ KOKKOS_ENABLE_OPTION(AGGRESSIVE_VECTORIZATION OFF "Whether to aggressively vecto
 KOKKOS_ENABLE_OPTION(LAUNCH_COMPILER      ON  "Whether to potentially use the launch compiler")
 KOKKOS_ENABLE_OPTION(COMPILE_AS_CMAKE_LANGUAGE OFF "Whether to use native cmake language support")
 KOKKOS_ENABLE_OPTION(HIP_MULTIPLE_KERNEL_INSTANTIATIONS OFF "Whether multiple kernels are instantiated at compile time - improve performance but increase compile time")
+KOKKOS_ENABLE_OPTION(HIP_UNIFIED_MEMORY OFF "Whether the HIPUnified memory space will be available")
```

Comment:
```suggestion
KOKKOS_ENABLE_OPTION(HIP_UNIFIED_MEMORY OFF "Whether the HIP unified memory space will be available")
```

Reply:
Your suggestion is closer to what was done with CudaUVM. I tried to directly use the (current but not final) name of the memory space as it is in the c++ code in an attempt to make it more transparent for users. Would you mind explaining why you would split the name into a description?

Reply:
This was mostly just for the consistency with other options. Seeing that most of our cmake options attempt to provide general descriptions without necessarily exposing in-code class or fn names, I was thinking that's what the users might prefer to see when running something like `cmake -LH`. But I admit that `HIPUnified` is self-explanatory enough that this suggestion wouldn't make much difference in this context.